README.md

```diff
@@ -72,28 +80,90 @@ This template follows a very simple project structure:
 - `index.js`: Entry point of your application as per React-Native standards.
 
 ## Splash screen customization
+
 To customize the splash screen (logo and background color) use the CLI provided in the [official docs](https://github.com/zoontek/react-native-bootsplash#assets-generation).
 
 ## Setup environments
+
 Modify the environment variables files in root folder (`.env.development`, `.env.production` and `.env.staging`)
 
 #### Android
+
 A map associating builds with env files is already defined in `app/build.gradle` you can modify or add environments if needed.
 
 For multiple enviroments to work you would need to change `-keep class [YOUR_PACKAGE_NAME].BuildConfig { *; }` on `proguard-rules.pro` file.
 
 #### iOS
+
 The basic idea in iOS is to have one scheme per environment file, so you can easily alternate between them.
 
 To create a new scheme:
+
 - In the Xcode menu, go to Product > Scheme > Edit Scheme
 - Click Duplicate Scheme on the bottom
 - Give it a proper name on the top left. For instance: "qa"
 - Then edit the newly created scheme to make it use a different env file. From the same "manage scheme" window:
 
 Expand the "Build" settings on left
+
 - Click "Pre-actions", and under the plus sign select "New Run Script Action"
 - Where it says "Type a script or drag a script file", type: `echo ".env.qa" > /tmp/envfile` replacing `.env.qa` with your file.
 
 ## Styleguide
+
 For coding styling we decided to go with ESLint and [React Native community's styleguide](https://github.com/facebook/react-native/tree/master/packages/eslint-config-react-native-community#readme).
+
+# How to use it
+
+The idea of this section is to explain how the template composition is better and easier to use when you try to use well formed arquitectures, especially using redux flow.
+
+First keep in mind that you will see the same way of classification in multiple places of the app. Different resources orchestrated by an index.js that allow you to call it as a single instance. This structure lets you use a simple way to import things.
+
+With that in mind we are going to look at each folder to explain how to use it.
+
+## Components
+
+The components are the basic structure of an application in react native, but specifically for this template the idea of ​​keeping the complexity of the development to a minimum, for this the ideal is that all the components are at the same nesting folder level. A component is only created if a specific view or functionality is being worked on or if the same object is to be reused in multiple parts of the application.
```

Comment:
Let's try not to be dogmatic here. When projects scale, this folder always ends up with some kind of nesting or internal structure anyway.

```suggestion
Components are the basic blocks of a react native application, but since we​​ aim to minimize development complexity, all the components are at the same nesting level.
```